### PR TITLE
Fix: HTTP network filters

### PR DESCRIPTION
**Pull request by aymericDD**
## What does this PR do?

- [ ] Adds new functionality
- [ ] Alters existing functionality
- [x] Fixes a bug
- [ ] Improves documentation or testing

Please briefly describe your changes as well as the motivation behind them:
- The http filters are not interpreted and disrupt all the requests. This fix updates the eBPF map correctly to ensure the custom paths and methods are taken into account.
- Update the documentation
- Improve performance by adding the directive `#pragma unroll` to instruct the compiler to unroll loops.  Loop unrolling is an optimization technique that can improve performance by reducing the overhead of loop control instruction-level  parallelism.

## Code Quality Checklist

- [x] The documentation is up to date.
- [x] My code is sufficiently commented and passes continuous integration checks.
- [x] I have signed my commit (see [Contributing Docs](../CONTRIBUTING.md)).

## Testing

- [ ] I leveraged continuous integration testing
    - [ ] by depending on existing `unit` tests or `end-to-end` tests.
    - [ ] by adding new `unit` tests or `end-to-end` tests.
- [x] I manually tested the following steps:
    - `x`
    - [x] locally.
    - [ ] as a canary deployment to a cluster.

### CODE REVIEW

File: ebpf/network-tc-filter/injection.bpf.c [hunk 1]
```diff
@@ -66,14 +80,26 @@ static __always_inline bool  validate_method(char* method) {
...
+        }
+
+        // No need to compare method if the expected method is empty
+        if (expected_method[0] == '\0')
+            break;
+
         // Check if the prefix match the method.
-        for (int i = 0; i < MAX_METHOD_LEN; ++i) {
+        #pragma unroll
+        for (int ii = 0; ii < MAX_METHOD_LEN; ++ii) {
```
Comment (clairecng) [inline ebpf/network-tc-filter/injection.bpf.c:95]: why ii?

File: ebpf/network-tc-filter/injection.bpf.c [hunk 2]
```diff
@@ -56,6 +69,7 @@ static __always_inline bool  validate_path(char* path) {
 }
 
 static __always_inline bool  validate_method(char* method) {
+     #pragma unroll
```
Comment (clairecng) [inline ebpf/network-tc-filter/injection.bpf.c:72]: what does this do

File: ebpf/network-tc-filter/injection.bpf.c [hunk 3]
```diff
@@ -134,10 +161,11 @@ int cls_classifier_paths(struct __sk_buff *skb)
         p[i] = load_byte(skb, skb_info.data_off + i);
     }
 
-    char path[MAX_PATH_LEN];
+    char path[MAX_PATH_LEN] = {0};
```
Comment (clairecng) [inline ebpf/network-tc-filter/injection.bpf.c:164]: what does this do? Seems a bit weird to me, why not initializing to empty string.
Each empty string should end by \0 which I think was your objective here?

```suggestion
    char path[MAX_PATH_LEN] = "";
```

Comment (aymericDD) [reply ebpf/network-tc-filter/injection.bpf.c:164]: Both char path[MAX_PATH_LEN] = {0}; and char path[MAX_PATH_LEN] = ""; achieve the same result of initializing the array with null characters (\0). However, char path[MAX_PATH_LEN] = {0}; is more explicit and commonly used in C programming to ensure all elements are set to 0.  

Here's a brief comparison:  
char path[MAX_PATH_LEN] = {0};: Explicitly initializes all elements of the array to 0.
char path[MAX_PATH_LEN] = "";: Initializes the first element to 0 and relies on the compiler to zero-initialize the rest.

Using char path[MAX_PATH_LEN] = {0}; is generally preferred for clarity and explicitness.

Comment (clairecng) [reply ebpf/network-tc-filter/injection.bpf.c:164]: gotcha, i disagree with copilot to me it's esotheric lol. Can you add a comment?

Comment (aymericDD) [reply ebpf/network-tc-filter/injection.bpf.c:164]: I have no preference, I can use `""` if you prefere 👍

Comment (clairecng) [reply ebpf/network-tc-filter/injection.bpf.c:164]: 🤷‍♀️ I have no preference with this explanation, it seems like it's a better practice to {0} so let's go with that.

Comment (aymericDD) [reply ebpf/network-tc-filter/injection.bpf.c:164]: After all, I prefer to use a simple `""`. I think it make more sense to use `{0}` when you are initialising a list.

File: ebpf/network-tc-filter/injection.bpf.c [hunk 4]
```diff
@@ -37,17 +38,29 @@ static __always_inline bool  validate_path(char* path) {
...
+        if (expected_path[0] == '\0')
+            break;
+
         char request_path[MAX_PATH_LEN];
         bpf_probe_read_kernel_str(&request_path, sizeof(request_path), path);
 
         // Check if the prefix match the method.
-        for (int i = 0; i < MAX_PATH_LEN; ++i) {
+        #pragma unroll
+        for (int jj = 0; jj < MAX_PATH_LEN; ++jj) {
```
Comment (clairecng) [inline ebpf/network-tc-filter/injection.bpf.c:57]: why double j? Why not just j?

Comment (aymericDD) [reply ebpf/network-tc-filter/injection.bpf.c:57]: Oups 😇

File: ebpf/network-tc-filter/injection.bpf.c [hunk 5]
```diff
@@ -66,14 +80,26 @@ static __always_inline bool  validate_method(char* method) {
...
+        }
+
+        // No need to compare method if the expected method is empty
+        if (expected_method[0] == '\0')
+            break;
+
         // Check if the prefix match the method.
-        for (int i = 0; i < MAX_METHOD_LEN; ++i) {
+        #pragma unroll
+        for (int jj = 0; jj < MAX_METHOD_LEN; ++jj) {
```
Comment (clairecng) [inline ebpf/network-tc-filter/injection.bpf.c:95]: this one as well 😆 sorry Aymeric

Comment (aymericDD) [reply ebpf/network-tc-filter/injection.bpf.c:95]: My bad 🙈

**Review by clairecng — APPROVED** [review]

**Pull request merged**